src/main/scala/model/mutation/Mutation.scala

```diff
@@ -0,0 +1,39 @@
+package model.mutation
+
+import model.genome.Genes
+import model.genome.Genes.GeneKind
+
+/**
+ * Model representation of the mutation
+ */
+trait Mutation {
+  val geneKind: GeneKind
+  val isDominant: Boolean
+  // add when introducing factors in the environments
+//  val factorsInfluenced: Factors
+//  def factorImpact()
+}
+
+object Mutation {
+  def apply(geneKind: GeneKind, isDominant: Boolean): Mutation = {
+    MutationImpl(geneKind, isDominant)
+  }
+
+  private case class MutationImpl( override val geneKind: GeneKind,
+                                  override val isDominant: Boolean ) extends Mutation
+}
+
+object Mutations extends Enumeration {
```

Comment:
Secondo me questa Enumeration è inutile, gli oggetti mutation sono creati dalla Gui e passati al Controller per poi aggiungerli dinamicamente all'Ambiente. Al massimo si potrebbe fare una Factory con un Object

Reply:
Anche secondo me questa Enumeration non serve (è un po' una ripetizione). Per indicare quale Allele è dominante e quale recessivo per ogni Gene, possiamo usare usare la funzione `setAlleleDominance` che trovi in Kinds, che assegna la dominanza un'unica volta. Sempre in Kinds potresti farti una funzione tipo `getDominantAllele` che dato un Gene ti ritorna l'Allele dominante, a questo punto ti basta usare l'enumeration Gene e questa funzione per ottenere un concetto simile all'enumeration mutations.

Reply:
Si non ero convintissima quando ho fatto l'enumeration se potesse tornare utile in futuro o meno. Per ora direi che la possiamo togliere.